### PR TITLE
Adding cultural check to the "ToUpperCase" string prototype.

### PR DESCRIPTION
**Pull request by LuisMerinoP**
Adding some easy win test fixes. 
Before:
image
With fix:
image

Comments:

- Lithuanian case is quite tricky. Only fixed the tests. Maybe there is some more work to do to check larger texts are upper-case-ed correctly.
- Maybe some very specific functionality in internationalization (such as `LithuanianStringProcessor`) needs its place in the project, kind of some utils folder or similar. Please advice if that needs to be moved somewhere else. 
- I only wanted to fix "ToLocaleUpperCase" tests because it seemed an easy win. Apparently, more tests get fixed. Don't know why and I did not dig into that.

### CODE REVIEW

**Comment by LuisMerinoP** [comment]
> But it seems there are other edge cases for tr, el and az

All but "el" cases are fixed on this PR, because there are tests for those: https://github.com/tc39/test262/tree/main/test/intl402/String/prototype/toLocaleUpperCase.

Greeks will need to wait :).

**Comment by lahma** [comment]
What worries me here is that this seems to just target the test case. Like input `"Hello I\u0307".toLocaleUpperCase("lt")` won't behave the same way with Jint and NodeJS.

**Comment by LuisMerinoP** [comment]
Yeah totally. Thanks for noticing. I focused on fixing the tests only that are character based and the Lithuanian case took all of my bandwidth, lol. That is what I meant with the 1st point of the PR description.
I will check the behaviour for larger strings and fix when I find the time :)

**Comment by LuisMerinoP** [comment]
Arranged the character filtering for larger strings. I can add some tests in [StringTests](https://github.com/sebastienros/jint/blob/main/Jint.Tests/Runtime/StringTests.cs) to verify corner cases and quirky scenarios provide the same output as node if that is useful. Let me know if that sounds good.
On the other hand, macos check is failing, don't know why.
Cheers :).

**Comment by LuisMerinoP** [comment]
> The more tests the better

Perfect, I'll add some.

**Comment by LuisMerinoP** [comment]
Adding tests for the string processing for all the special character cases.

**Comment by lahma** [comment]
So is this ready for merge or still some work pending?

**Comment by LuisMerinoP** [comment]
According to me, its ready for review and eventual merge :)

File: Jint/Native/String/StringInlHelper.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
+﻿using System.Text;
+
+namespace Jint.Native.String
```
Comment (lahma) [inline Jint/Native/String/StringInlHelper.cs:3]: nit: for new files should use file-scoped namespaces

File: Jint/Native/String/StringInlHelper.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+{
+    /// <summary>
+    /// Some internacionalization logic that is special or specific to determined culture.
+    /// </summary>
+    internal class StringInlHelper
```
Comment (lahma) [inline Jint/Native/String/StringInlHelper.cs:8]: nit: `internal` types should be either sealed or static

File: Jint/Native/String/StringInlHelper.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        /// <summary>
+        /// Lithuanian case is a bit special. For more info see:
+        /// https://github.com/tc39/test262/blob/main/test/intl402/String/prototype/toLocaleUpperCase/special_casing_Lithuanian.js
+        /// </summary>
+        public static string LithuanianStringProcessor(string input)
+        {
+            var replaceableCharsIdx = GetLithuaninanReplaceableCharIdx(input);
+            if (replaceableCharsIdx.Count > 0)
+            {
+                StringBuilder stringBuilder = new StringBuilder(input);
```
Comment (lahma) [inline Jint/Native/String/StringInlHelper.cs:37]: we have ValueStringBuilder for building these

File: Jint/Native/String/StringPrototype.cs [hunk 4]
```diff
@@ -231,7 +232,32 @@ private JsValue ToLocaleUpperCase(JsValue thisObject, JsValue[] arguments)
         {
             TypeConverter.CheckObjectCoercible(_engine, thisObject);
             var s = TypeConverter.ToString(thisObject);
-            return new JsString(s.ToUpper(CultureInfo.InvariantCulture));
+            var culture = CultureInfo.InvariantCulture;
+            if (arguments.Length > 0 && arguments[0].IsString())
```
Comment (lahma) [inline Jint/Native/String/StringPrototype.cs:236]: migth be cleaner to use `arguments.At(0) is JsString s`

**Review by lahma — APPROVED** [review]
Added some remarks, but as not in hot path we don't need to worry about them that much.

**Comment by lahma** [comment]
Thank you for your contribution! At some point probably would be a good idea to enable some tests against these features.
